wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -176,22 +174,34 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
         }
     }
 
-    fun isPlatformAvailable(): Resource<Boolean> {
-        // this checks only one random node, but will retry 10 times.
-        // it is possible that some nodes are not available due to location,
-        // firewalls or other reasons
-        return try {
-            val response = platform.client.getStatus()
-            Resource.success(response!!.connections > 0 && /*response.errors.isBlank() &&*/
-                    Constants.NETWORK_PARAMETERS.getProtocolVersionNum(NetworkParameters.ProtocolVersion.MINIMUM) <= response.protocolVersion)
-        } catch (e: Exception) {
-            Resource.error(e.localizedMessage!!, null)
+    /**
+     * Calls Platform.check() three times asynchronously
+     *
+     * @return true if platform is available
+     */
+    suspend fun isPlatformAvailable(): Resource<Boolean> {
+        return withContext (Dispatchers.IO) {
+            var success = 0
+            val checks = arrayListOf<Deferred<Boolean>>()
+            for (i in 0 until 3) {
+                checks.add(async { platform.check() })
+            }
+
+            for (check in checks) {
+                success += if (check.await()) 1 else 0
+            }
```

Comment:
Here, we are using co-routines to call one function three times asynchronously with `async`.

`await()` is used to wait till that function finishes.

Reply:
Nice! 👍